xrpl/models/utils.py

```diff
@@ -1,9 +1,18 @@
 """Helper util functions for the models module."""
+
+import re
 from dataclasses import dataclass, is_dataclass
-from typing import Any, Dict, List, Type, TypeVar, cast
+from typing import Any, Dict, List, Pattern, Type, TypeVar, cast
+
+from typing_extensions import Final
 
 from xrpl.models.exceptions import XRPLModelException
 
+# Regular-Expression pertaining to Credentials Ledger Object transactions
+# Note: This regex is not identical to the one used with DIDSet transaction. This regex
+# mandates a minimum of length-1 strings.
+HEX_REGEX: Final[Pattern[str]] = re.compile("[a-fA-F0-9]+")
```

Comment:
Why not use the same hex regex for both DID strings and credential strings and just add an additional check to ensure the string isn't empty for credentials? Then it's more future-proof too (and much less confusing).

Reply:
The restrictions on hexa-decimal format and non-zero are applicable to both `uri` and `credential_type` fields. If we split them into two separate checks (as you propose), an engineer might forget to include the second length check explicity.

I feel using separate regex-es is still a better solution. I can rename this regex to something more appropriate, `CREDENTIAL_REGEX` ?

Reply:
I think it's less confusing to require the check than to only allow this regex to be usable in _some_ hex situations.

Reply:
what is the rationale for the difference in the behavior between DID and Credentials strings?

Reply:
In DID, an empty string in a transaction deletes the field. That's not relevant for credentials.

Reply:
I've incorporated your suggestion in e63b49d4